presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -165,24 +166,27 @@ public GlueHiveMetastore(
             HdfsEnvironment hdfsEnvironment,
             GlueHiveMetastoreConfig glueConfig,
             GlueColumnStatisticsProvider columnStatisticsProvider,
-            @ForGlueHiveMetastore Executor executor)
+            @ForGlueHiveMetastore Executor executor,
+            @ForGlueHiveMetastore Optional<RequestHandler2> requestHandler)
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.hdfsContext = new HdfsContext(ConnectorIdentity.ofUser(DEFAULT_METASTORE_USER));
-        this.glueClient = requireNonNull(createAsyncGlueClient(glueConfig), "glueClient is null");
+        this.glueClient = requireNonNull(createAsyncGlueClient(glueConfig, requestHandler), "glueClient is null");
```

Comment:
Let's just remove this null check. We can trust that `createAsyncGlueClient` doesn't return null since it's private code to this class.